sheepdog/transactions/deletion/transaction.py

```diff
@@ -177,3 +182,12 @@ def get_nodes(self, ids):
                 } for c in node._related_cases_from_cache]
                 for node in nodes
             }
+
+    def mark_indexd_delete(self, ids):
```

Comment:
I think you should at least retry once and then raise an exception if failure continues. Retry within the loop, so as to ensure one failure does not override all the hard work done so far

Reply:
For error handling aspect, this is after nodes are committed.  A failure here will have some or all indexd doc not have deleted flag, while corresponding nodes are deleted. This is impossible to handle given indexd has no transaction support.

I see your point where we add retries, simply due to the unreliable nature of indexd.